src/ipyradiant/loader/manager.py

```diff
@@ -0,0 +1,76 @@
+""" loading widgets
+"""
+# Copyright (c) 2020 ipyradiant contributors.
+# Distributed under the terms of the Modified BSD License.
+
+import traitlets as T
+
+import ipywidgets as W
+from rdflib import BNode, Graph
+
+from .base import BaseLoader
+from .upload import UpLoader
+from .util import get_n_predicates, get_n_subjects
+
+
+class FileManager(W.VBox):
+    """ Wraps a file selector and stats
+    """
+
+    n_triples = T.Int()
+    n_subjects = T.Int()
+    n_predicates = T.Int()
+
+    loader = T.Instance(BaseLoader)
```

Comment:
Sanity check here for me. You set this as BaseLoader because the FileManager could use any type of loader in this Widget, but you set the default on line 40 as UpLoader because the current version of this FileManager widget uses the Uploader as the loader attr. Is that correct? You __could__ have said this line to `T.Instance(UpLoader)`, but that would be __less__ maintainable because it would require you to change two lines if we ever used a different type of loader for this widget?

Reply:
right. it should only use the API we define in `BaseLoader`.